src/metadata/audiotagger.cpp

```diff
@@ -0,0 +1,107 @@
+#include "metadata/audiotagger.h"
+
+#include "metadata/trackmetadatataglib.h"
+
+#include "util/assert.h"
+
+#include <taglib/vorbisfile.h>
+#include <taglib/wavpackfile.h>
+#include <taglib/mpegfile.h>
+#include <taglib/mp4file.h>
+#include <taglib/oggfile.h>
+#include <taglib/flacfile.h>
+#include <taglib/aifffile.h>
+#include <taglib/rifffile.h>
+#include <taglib/wavfile.h>
+#if (TAGLIB_MAJOR_VERSION > 1) || ((TAGLIB_MAJOR_VERSION == 1) && (TAGLIB_MINOR_VERSION >= 9))
+#include <taglib/opusfile.h>
+#endif
+
+#include <QtDebug>
+
+namespace {
+
+inline SecurityTokenPointer openSecurityToken(QFileInfo file,
+        const SecurityTokenPointer& pToken) {
+    if (pToken.isNull()) {
+        return Sandbox::openSecurityToken(file, true);
+    } else {
+        return pToken;
+    }
+}
+
+} // anonymous namespace
+
+AudioTagger::AudioTagger(const QString& file, SecurityTokenPointer pToken) :
+        m_file(file),
+        m_pSecurityToken(openSecurityToken(m_file, pToken)) {
+}
+
+AudioTagger::~AudioTagger() {
+}
+
+bool AudioTagger::save(const Mixxx::TrackMetadata& trackMetadata) {
```

Comment:
Info: 
I do not know if you have followed the discussion why this code is disabled. 
https://bugs.launchpad.net/mixxx/+bug/728197
Issues: 
* General stability concerns  (segfaults due to corrupt files ..) 
* Taglib uses in place writing without a backup file. 
* Adding a Tag will shift samples and may produces decoding errors. 
I have stated to add a quarantine process adopted from Clementine, but I have not managed to finish it.